runtime/recipe.js

```diff
@@ -52,6 +52,18 @@ class RecipeComponent {
       arc.connectParticleToView(particle, connection.name, connection.view);
     }
   }
+
+  findConnectionByName(name) {
+    for (let connection of this.connections)
+      if (connection.name == name)
+        return connection;
+  }
+
+  findConnectionByConstraintName(constraintName) {
+    for (let connection of this.connections)
+      if (connection.constraintName == constraintName)
```

Comment:
Suggestion using `===` (triple-equal) by default, unless you really want loose comparison.